homeassistant/components/mqtt/image.py

```diff
@@ -0,0 +1,229 @@
+"""Support for MQTT images."""
+from __future__ import annotations
+
+from base64 import b64decode
+import binascii
+from collections.abc import Callable
+import functools
+import logging
+import ssl
+from typing import Any
+
+import httpx
+import voluptuous as vol
+
+from homeassistant.components import image
+from homeassistant.components.image import (
+    DEFAULT_CONTENT_TYPE,
+    ImageEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, CONF_VALUE_TEMPLATE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.service_info.mqtt import ReceivePayloadType
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.util import dt as dt_util
+
+from . import subscription
+from .config import MQTT_BASE_SCHEMA
+from .const import CONF_ENCODING, CONF_QOS, CONF_TOPIC
+from .debug_info import log_messages
+from .mixins import MQTT_ENTITY_COMMON_SCHEMA, MqttEntity, async_setup_entry_helper
+from .models import MessageCallbackType, MqttValueTemplate, ReceiveMessage
+from .util import get_mqtt_data, valid_subscribe_topic
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CONTENT_TYPE = "content_type"
+CONF_IMAGE_ENCODING = "image_encoding"
+CONF_FROM_URL_TOPIC = "from_url_topic"
+
+DEFAULT_NAME = "MQTT Image"
+
+
+def validate_topic_required(config: ConfigType) -> ConfigType:
+    """Ensure at least one subscribe topic is configured."""
+    if CONF_TOPIC not in config and CONF_FROM_URL_TOPIC not in config:
+        raise vol.Invalid("Expected one of [`topic`, `from_url_topic`], got none")
+    if CONF_CONTENT_TYPE in config and CONF_FROM_URL_TOPIC in config:
+        raise vol.Invalid(
+            "Option `content_type` and not be used together with `from_url_topic`"
+        )
+    return config
+
+
+PLATFORM_SCHEMA_BASE = MQTT_BASE_SCHEMA.extend(
+    {
+        vol.Optional(CONF_CONTENT_TYPE): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Exclusive(CONF_FROM_URL_TOPIC, "image_topic"): valid_subscribe_topic,
+        vol.Exclusive(CONF_TOPIC, "image_topic"): valid_subscribe_topic,
+        vol.Optional(CONF_IMAGE_ENCODING): "b64",
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
```

Comment:
If this is only supported to extract the URL, I think it should be `url_template` instead